relayer-engine/src/walletMonitor/walletMonitorHarness.ts

```diff
@@ -0,0 +1,86 @@
+import { CHAIN_ID_SOLANA, isEVMChain, isTerraChain } from "@certusone/wormhole-sdk";
+import { ChainConfigInfo } from "../../packages/relayer-plugin-interface";
+import { getCommonEnv, CommonEnv, ExecutorEnv, getExecutorEnv, PrivateKeys, TokenMonitoringInfo } from "../config";
+import { getScopedLogger, ScopedLogger } from "../helpers/logHelper";
+import { sleep } from "../utils/utils"
+import { WalletBalance, pullAllSolanaBalances, pullAllEVMBalances, pullAllTerraBalances } from './balances';
+import { updateWalletBalances } from "./metrics";
+
+const ONE_MINUTE: number = 60000;
+
+let _logger: ScopedLogger;
+
+const logger = () => {
+  if (!_logger) _logger = getScopedLogger(["WalletMonitor"]);
+  return _logger;
+}
+
+export type ChainMonitoringConfig = {
+  chainInfo: ChainConfigInfo,
+  privateKeys: string[],
```

Comment:
Actually you don't need them. It was done this way because it was tied to the executor in a way... I mean, balances were monitored for all private-keys in the executor config as a way to monitor all relevant wallets... bur really it would be ideal to have a separate config (specially if we'll make it a separate module)
I think we can fix this when we port the wallet-monitor to it's own repo this week, would you agree?

Reply:
i meant that you obtain the public key further inside the code, you could refactor that happen outside